src/wallet/init.cpp

```diff
@@ -120,6 +120,12 @@ bool WalletInit::ParameterInteraction() const
         LogPrintf("%s: parameter interaction: -blocksonly=1 -> setting -walletbroadcast=0\n", __func__);
     }
 
+    if (gArgs.GetBoolArg("-privatebroadcast", DEFAULT_PRIVATE_BROADCAST) &&
```

Comment:
Yes, but I did not like the behavior of the wallet "unexpectedly" stopping to broadcast its own transactions because `-privatebroadcast` is enabled.

> It's not the most user-friendly thing to introduce a new command arg that would only work if another arg was also changed from its default.

Yes, I agree. I do not like that either but somewhat prefer it over automatically switching off wallet broadcast. I can imagine GUI users who enable `-privatebroadcast` and still try to send transactions from their wallets and have no clue why they are not being mined.

Leaving this open to discussion and proposals.

Reply:
I wonder if the option should be renamed `-privaterpcbroadcast` for now, and we can add `-privatewalletbroadcast` later. 

We also won't be able to prevent users from using their wallet with `sendrawtransaction` anyway. So if the goal here is to prevent accidental privacy leaks from confused users, we might want to consider "hiding" the option or restricting it to test networks until wallet integration is complete.

Reply:
I thought about this further and removed the requirement to set `-walletbroadcast=0` when `-privatebroadcast=1`. The description of `-privatebroadcast` begins with

> Broadcast transactions submitted via sendrawtransaction RPC using short lived connections...

so it should be clear that this applies to the `sendrawtransaction` RPC. In addition I extended the text with

> Transactions submitted through the wallet are not affected by this option

Reply:
Marking this as resolved because I removed the requirement to set `-walletbroadcast=0` and extended the `-privatebroadcast` doc.

> My understanding was that anything done via the wallet would work as usual, but that transactions sent via sendrawtransaction would use the private broadcast mechanism if set

Yes, this is exactly how it works now.